server/app/services/applicant/ApplicantService.java

```diff
@@ -635,35 +647,51 @@ private CompletionStage<Void> maybeNotifyTiSubmitter(
         .toCompletableFuture();
   }
 
+  /**
+   * Email the applicant with either the default application email or the status' defined email.
+   * Will send nothing if the applicant's email address doesn't exist. Will send the default
+   * application email if the status does not have an email body defined.
+   *
+   * @param applicantId the ID of the applicant
+   * @param applicationId the ID of the application
+   * @param programDef the ProgramDefinition that the applicant applied for
+   * @param status the status from which to get the email body to send
+   */
   private CompletionStage<Void> maybeNotifyApplicant(
-      long applicantId, long applicationId, String programName) {
+      long applicantId,
+      long applicationId,
+      ProgramDefinition programDef,
+      Optional<StatusDefinitions.Status> status) {
     CompletableFuture<Optional<String>> emailFuture = getEmail(applicantId).toCompletableFuture();
     CompletableFuture<Optional<Locale>> localeFuture =
         getPreferredLocale(applicantId).toCompletableFuture();
     return CompletableFuture.allOf(emailFuture, localeFuture)
         .thenRunAsync(
             () -> {
-              // Not blocking since these are already completed
-              Optional<String> email = emailFuture.join();
               Locale locale = localeFuture.join().orElse(LocalizedStrings.DEFAULT_LOCALE);
+              Optional<String> email = emailFuture.join();
```

Comment:
Could we just get the email before this async block then return early before the existing return statement? Then we will have a `String` ready to go for the async block

Reply:
I find it more readable, but if you disagree feel free to roll it back; I won't block on it (hehe, `CompletableFuture` pun)